hoomd/state.py

```diff
@@ -38,33 +39,166 @@ def _create_domain_decomposition(device, box):
+    Each of these data structures is similar, differing in the number of
+    particles in the group and what operations use them. Bonds, angles,
+    dihedrals, and impropers contain 2, 3, 4, and 4 particles per group
+    respectively. Bonds specify the toplogy used when computing energies and
+    forces in `md.bond`, angles define the same for `md.angle`, dihedrals for
+    `md.dihedral` and impropers for `md.improper`. These collectively implement
+    bonding potentials used in molecular dynamics force fields. Like bonds,
+    special pairs define connections between two particles, but special pairs
+    are intended to adjust the 1-4 pairwise interactions in some molecular
+    dynamics force fields: see `md.special_pair`. Each bonded group is defined
+    by a type id, the group members, and a tag.
+
+    - :math:`t_\\mathrm{id}`: type id :math:`[\\mathrm{dimensionless}]` -
+      integer in the range ``[0,len(bond_types)``) that identifies the
+      bond's type. `bond_types` maps type ids to names with:
+      ``name = particle_types[t_id]``. Similarly, `angle_types` lists the
+      angle types, `dihedral_types` lists the dihedral types, `improper_types`
+      lists the improper types, and `special_pair_types` lists the special pair
+      types.
+    - group members: a list of integers in the range ``[0,max(p_tag)]`` that
+      defines the tags of the particles in the bond (2), angle (3), dihedral
+      (4), improper (4), or special pair (2).
+    - :math:`b_\\mathrm{tag}`` : tag :math:`[\\mathrm{dimensionless}]` -
+      integer that uniquely identifies a given bond. The bonds are in
+      tag order when writing and initializing to/from a GSD file or snapshot
+      :math:`b_\\mathrm{tag,i} = i`. When accessing data in local snapshots,
+      bonds may be in any order.
+
+    .. rubric:: Constraints
+
+    The state contains `N_constraints` distance constraints between particles.
+    These constraints are enforced by `md.constrain.Distance`. Each distance
+    constraint consists of a distance value and the group members.
+
+    - group members: a list of 2 integers in the range ``[0,max(p_tag)]`` that
+      defines the tags of the particles in the constraint.
+    - :math:`d`: constraint value :math:`[\\mathrm{length}]` - the distance
+      between particles in the constraint.
+
+    .. rubric:: MPI
+
+    When running in serial or on 1 MPI rank, the entire simulation state is
+    stored in that process. When using more than 1 MPI rank, HOOMD-blue employs
+    a domain decomposition approach to split the simulation box an integer
+    number of times in the x, y, and z directions. Each MPI rank stores and
+    operates on the particles local to that rank, those contained within the
+    region defined by the split planes. Each MPI rank communicates with its
+    neighbors to obtain the properties of particles (ghost particles) near the
+    boundary between ranks so that it can compute interactions across the
+    boundary.
```

Comment:
I moved the name "ghost particles" so it's clearer that it corresponds to the phrase "particles near the boundary between ranks."
```suggestion
    neighbors to obtain the properties of particles near the boundary between
    ranks (ghost particles) so that it can compute interactions across the
    boundary.
```